pkg/services/ngalert/provisioning/contactpoints.go

```diff
@@ -49,10 +51,22 @@ type ContactPointQuery struct {
 	Decrypt bool
 }
 
+func (ecp *ContactPointService) canDecryptSecrets(ctx context.Context, u *user.SignedInUser) bool {
+	if u == nil {
+		return false
+	}
+	permitted, err := ecp.ac.Evaluate(ctx, u, accesscontrol.EvalPermission(accesscontrol.ActionAlertingProvisioningReadSecrets))
+	if err != nil {
+		ecp.log.Error("Failed to evaluate user permissions", "error", err)
+		permitted = false
+	}
+	return permitted
+}
+
 // GetContactPoints returns contact points. If q.Decrypt is true and the user is an OrgAdmin, decrypted secure settings are included instead of redacted ones.
 func (ecp *ContactPointService) GetContactPoints(ctx context.Context, q ContactPointQuery, u *user.SignedInUser) ([]apimodels.EmbeddedContactPoint, error) {
-	if q.Decrypt && (u == nil || !u.HasRole(org.RoleAdmin)) {
-		return nil, fmt.Errorf("%w: decrypting secure settings requires Org Admin", ErrPermissionDenied)
+	if q.Decrypt && !ecp.canDecryptSecrets(ctx, u) {
+		return nil, fmt.Errorf("%w: user does not have permissions to read decrypted secure settings", ErrPermissionDenied)
```

Comment:
This will be shown to users, should we mention the specific permission necessary here? Something like: `user requires Org Admin or alert.provisioning.secrets:read permission to view decrypted secure settings`.

Reply:
I like the suggestion. I wanted to check for GE and returns different errors depending on it but then decided that it would be too much.